Error when building python wrapping: `SWIGPY_SLICEOBJECT` has not been declared

<!-- The text within this markup is a comment, and is intended to provide
guidelines to open an issue for the ITK repository. This text will not
be part of the issue. -->


<!-- Before submitting an issue, please check that your issue has not been
already filed. -->

### Description

When trying to build itk version 5.3.0 for openSUSE Tumbleweed (compiler and other library versions detailed below), we see errors about `SWIGPY_SLICEOBJECT` not being defined when generating the python bindings, leading to typecast errors from `int *` to `PyObject *`.


<!-- Description of the bug. -->

### Steps to Reproduce

1. Configure itk build with `-DITK_WRAP_PYTHON:BOOL=ON`; the full cmake command issued was:
    `/usr/bin/cmake /home/abuild/rpmbuild/BUILD/ITK-5.3.0/. -GNinja -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DCMAKE_INSTALL_LIBDIR:PATH=lib64 -DCMAKE_INSTALL_LIBEXECDIR=/usr/libexec -DCMAKE_BUILD_TYPE=RelWithDebInfo '-DCMAKE_C_FLAGS=-O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g' '-DCMAKE_CXX_FLAGS=-O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g' '-DCMAKE_Fortran_FLAGS=-O2 -Wall -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g' '-DCMAKE_EXE_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' '-DCMAKE_MODULE_LINKER_FLAGS=-flto=auto -Wl,--as-needed' '-DCMAKE_SHARED_LINKER_FLAGS=-flto=auto -Wl,--as-needed -Wl,--no-undefined -Wl,-z,now' -DLIB_SUFFIX=64 -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_STATIC_LIBS:BOOL=OFF -DCMAKE_COLOR_MAKEFILE:BOOL=OFF -DCMAKE_INSTALL_DO_STRIP:BOOL=OFF -DCMAKE_MODULES_INSTALL_DIR=/usr/lib64/cmake/insighttoolkit -DITK_INSTALL_LIBRARY_DIR:PATH=lib64/ -DITK_INSTALL_INCLUDE_DIR:PATH=include/insighttoolkit/ -DITK_INSTALL_PACKAGE_DIR:PATH=lib64/cmake/insighttoolkit/ -DITK_INSTALL_RUNTIME_DIR:PATH=/usr/bin -DITK_INSTALL_DOC_DIR=share/doc/packages/insighttoolkit/ -DBUILD_EXAMPLES:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=ON -DBUILD_TESTING:BOOL=OFF -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON -DITK_USE_FFTWD:BOOL=ON -DITK_USE_FFTWF:BOOL=ON -DITK_USE_SYSTEM_LIBRARIES:BOOL=ON -DITK_USE_SYSTEM_CASTXML:BOOL=ON -DITK_USE_SYSTEM_GDCM:BOOL=ON -DITK_USE_SYSTEM_SWIG:BOOL=ON -DITK_USE_SYSTEM_VXL:BOOL=OFF -DVXL_BUILD_CORE_NUMERICS:BOOL=OFF -DVCL_INCLUDE_CXX_0X:BOOL=ON -DITK_FORBID_DOWNLOADS=ON -DITK_WRAP_PYTHON:BOOL=ON`
1. Run `make -j<N>`

### Expected behavior

Python bindings to be generated, build to complete successfully.

### Actual behavior

```
[  542s] /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp:4942:149: error: 'SWIGPY_SLICEOBJECT' has not been declared
[  542s]  4942 | SWIGINTERN std::vector< itkPointF2,std::allocator< itkPointF2 > > *std_vector_Sl_itkPointF2_Sg____getitem____SWIG_0(std::vector< itkPointF2 > *self,SWIGPY_SLICEOBJECT *slice){
[  542s]       |                                                                                                                                                     ^~~~~~~~~~~~~~~~~~
[  542s] /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp: In function 'std::vector<itk::Point<float, 2>, std::allocator<itk::Point<float, 2> > >* std_vector_Sl_itkPointF2_Sg____getitem____SWIG_0(std::vector<itk::Point<float, 2>, std::allocator<itk::Point<float, 2> > >*, int*)':
[  542s] /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp:4948:26: error: cannot convert 'int*' to 'PyObject*' {aka '_object*'}
[  542s]  4948 |       PySlice_GetIndices(slice, (Py_ssize_t)self->size(), &i, &j, &step);
[  542s]       |                          ^~~~~
[  542s]       |                          |
[  542s]       |                          int*
[  542s] In file included from /usr/include/python3.10/Python.h:106,
[  542s]                  from /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp:170:
[  542s] /usr/include/python3.10/sliceobject.h:41:46: note:   initializing argument 1 of 'int PySlice_GetIndices(PyObject*, Py_ssize_t, Py_ssize_t*, Py_ssize_t*, Py_ssize_t*)'
[  542s]    41 | PyAPI_FUNC(int) PySlice_GetIndices(PyObject *r, Py_ssize_t length,
[  542s]       |                                    ~~~~~~~~~~^
[  542s] /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp: At global scope:
[  542s] /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp:4953:98: error: 'SWIGPY_SLICEOBJECT' has not been declared
[  542s]  4953 | SWIGINTERN void std_vector_Sl_itkPointF2_Sg____setitem____SWIG_0(std::vector< itkPointF2 > *self,SWIGPY_SLICEOBJECT *slice,std::vector< itkPointF2,std::allocator< itkPointF2 > > const &v){
[  542s]       |                                                                                                  ^~~~~~~~~~~~~~~~~~
[  542s] /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp: In function 'void std_vector_Sl_itkPointF2_Sg____setitem____SWIG_0(std::vector<itk::Point<float, 2>, std::allocator<itk::Point<float, 2> > >*, int*, const std::vector<itk::Point<float, 2>, std::allocator<itk::Point<float, 2> > >&)':
[  542s] /home/abuild/rpmbuild/BUILD/ITK-5.3.0/build/Wrapping/Modules/ITKCommon/itkPointPython.cpp:4959:26: error: cannot convert 'int*' to 'PyObject*' {aka '_object*'}
[  542s]  4959 |       PySlice_GetIndices(slice, (Py_ssize_t)self->size(), &i, &j, &step);
[  542s]       |                          ^~~~~
[  542s]       |                          |
[  542s]       |                          int*
[  542s] /usr/include/python3.10/sliceobject.h:41:46: note:   initializing argument 1 of 'int PySlice_GetIndices(PyObject*, Py_ssize_t, Py_ssize_t*, Py_ssize_t*, Py_ssize_t*)'
```

### Reproducibility

100%

### Versions

ITK version 5.3.0

### Environment

* OS: Linux (openSUSE Tumbleweed, x86_64, kernel 6.0.10)
* GCC: 12.2.1
* Python 3.10.8
* Swig: 4.1.0


### Additional Information

Full build log here: [insighttoolkit_build_log.zip](https://github.com/InsightSoftwareConsortium/ITK/files/10145736/insighttoolkit_build_log.zip)

--- a/Wrapping/Generators/Python/module_ext.i.in
+++ b/Wrapping/Generators/Python/module_ext.i.in
@@ -6,8 +6,10 @@
 
 #if PY_VERSION_HEX >= 0x03020000
 # define SWIGPY_SLICE_ARG(obj) ((PyObject*) (obj))
+# define SWIGPY_SLICEOBJECT PyObject
 #else
 # define SWIGPY_SLICE_ARG(obj) ((PySliceObject*) (obj))
+# define SWIGPY_SLICEOBJECT PySliceObject
 #endif
 %}
 

